### PR TITLE
chore: dependency cleanup

### PR DESCRIPTION
**Pull request by joshuarli**
Wanted to try cleaning up the `requirements` texts.

For every top level entry in `pipdeptree --reverse`, that is also specified in requirements*.txt and is not exclusively pulled in by sentry, is worth taking a look at.

These are:

	urllib3
    redis
    pycodestyle
	botocore
    lxml
    ipaddress
    futures
    functools32
	enum34
    cssutils
    six

At first, i'm just going to remove all of these and see what happens, even though we explicitly import some of them. Here's the before and after `pip freeze` diff:

```diff
12c12
< botocore==1.5.70
---
> botocore==1.5.95
26c26
< cssutils==0.9.10
---
> cssutils==1.0.2
118c118
< redis==2.10.5
---
> redis==3.3.8
141c141
< urllib3==1.24.2
---
> urllib3==1.24.3
```

Some remarks:

- urllib3, botocore should IMO be left to requests and boto3 to bump, respectively, so this is good, even if we do explictly import them in some places
- cssutils needs to be bumped to 1.0.2 anyways if we want to switch to python3: https://bitbucket.org/cthedot/cssutils/issues/52/bad-octal-escape-blows-up-on-python-35b3
- mainly concerned about the huge redis jump.

### CODE REVIEW

**Comment by joshuarli** [comment]
Also.

```
* google-cloud-storage==1.13.2
 - google-cloud-core [required: >=0.29.0,<0.30dev, installed: 1.0.3]
```

pip will complain about this: `ERROR: google-cloud-storage 1.13.2 has requirement google-cloud-core<0.30dev,>=0.29.0, but you'll have google-cloud-core 1.0.3 which is incompatible.`

@mattrobenolt any opinions on whether or not this should be fixed?

File: requirements-base.txt [hunk 1]
```diff
@@ -1,33 +1,23 @@
...
 croniter>=0.3.26,<0.4.0
-cssutils>=0.9.9,<0.10.0
 django-crispy-forms>=1.4.0,<1.5.0
 django-jsonfield>=0.9.13,<0.9.14
 django-picklefield>=0.3.0,<1.1.0
 django-sudo>=2.1.0,<3.0.0
 Django>=1.8,<1.9
 djangorestframework==3.0.5
 email-reply-parser>=0.2.0,<0.3.0
-enum34>=1.1.6,<1.2.0
```
Comment (wedamija) [inline requirements-base.txt]: We use this around the codebase a lot

File: requirements-base.txt [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
 exam>=0.5.1
-functools32>=3.2.3,<3.3
```
Comment (wedamija) [inline requirements-base.txt]: I can see this used as well

**Comment by joshuarli** [comment]
lxml is only used here, and it's imported explicitly by us but right next to toronado, which requires it, so I'm thinking it's fine to remove lxml from requirements.

```
src/sentry/utils/email.py
15:import lxml
45:    tree = lxml.html.document_fromstring(value)
49:    return lxml.html.tostring(tree, doctype="<!DOCTYPE html>")
```

**Comment by joshuarli** [comment]
Putting back the `enum34` and `functool32` backport packages.

**Comment by joshuarli** [comment]
`pycodestyle` and `cssutils` are the only ones not explicitly referenced, so those are completely fine to remove.

**Comment by mattrobenolt** [comment]
fwiw I really don't like this. We've had issues with transient dependencies not being compatible, specifically botocore has caused us problems in the past. I agree and understand in theory, but in practice, people aren't good at versioning enough that I trust this without extensive testing of every single thing.

**Comment by joshuarli** [comment]
@mattrobenolt Ah, ok. Makes sense.

I guess me working on this PR boils down to a few things:

- cssutils needs to be bumped to 1.0.2
- pycodestyle can/should be removed (https://github.com/getsentry/sentry/pull/14710)
- packages that are backports should be annotated with a `# TODO: Remove once we no longer support Python 2.7`

**Comment by joshuarli** [comment]
Another question - `cryptography` is commented out but we do explicity refer to it.

```
cryptography==2.7
  - pyOpenSSL==19.0.0 [requires: cryptography>=2.3]
  - python-u2flib-server==5.0.0 [requires: cryptography>=1.2]
    - sentry==10.0.0.dev0 [requires: python-u2flib-server>=5.0.0,<6.0.0]
```

Should we pin it? Or at least `cryptography` to make clear we explicitly use it.

**Pull request closed without merging**